specification/customer-insights/resource-manager/Microsoft.CustomerInsights/2017-01-01/customer-insights.json

```diff
@@ -488,16 +535,822 @@
             "in": "path",
             "required": true,
             "type": "string",
-            "description": "The name of the hub."
+            "description": "The name of the Hub.",
+            "pattern": "^[a-zA-Z][a-zA-Z0-9]+$",
+            "minLength": 1,
+            "maxLength": 64
           },
           {
-            "name": "profileName",
-            "in": "path",
+            "name": "parameters",
+            "in": "body",
             "required": true,
-            "type": "string",
-            "description": "The name of the profile."
-          },
-          {
+            "schema": {
+              "$ref": "#/definitions/Hub"
+            },
+            "description": "Parameters supplied to the CreateOrUpdate Hub operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```

Comment:
Model a "default" response for all the APIs which captures the error cases and the response is compliant with ARM RPC error contract. Example here - https://github.com/Azure/azure-rest-api-specs/blob/current/specification/batch/resource-manager/Microsoft.Batch/2017-05-01/BatchManagement.json#L149